### ISSUE TITLE
PiHome version 2.0

### ISSUE THREAD

**Issue by pihome-shc** [pihome-shc/pihome#416]
i was thinking should we aim for version 2.0 now, at version 2.0 we will drop upgrade from previous version. 
All extra features updates goes toward version 2.0

what you guys think?

**Comment by sandreialexandru** [comment]
In my opinion, going to version 2, which is a big step, should include some major improvements and bug fixes.
#393 #415 #413 #417 are the ones that i'm thinking about, but I may be subjective 😁

**Comment by pihome-shc** [comment]
@sandreialexandru  yes move from 1.x version to 2.x  will have some upgrades and we need to decide database changes and try to stick with them so we don't have to update database or at least no major db update. in version 2.x

**Comment by twa127** [comment]
Hi,

Sounds good to me.

Some of the changes are fundamental to how the system operates, I see version 2 moving away from being zone based eg #393 this will require some significant changes to the database but should add more flexibility for future enhancements. 

#393 is almost ready to go, perhaps this should be the first version 2 release

**Comment by sandreialexandru** [comment]
Having in mind that no future upgrades to the database should be made, I would suggest taking into consideration #413 . Doesn't anyone find this implementation useful? Something like this
![6426206700041-termostat-de-camera-electronic-digital-cu-fir-ct-3s-1042](https://user-images.githubusercontent.com/45868553/96348063-b0b5f200-10ae-11eb-9db5-9f60acbec987.jpg)

Maybe running a few tests with the reset button should be done before v2. I'm still thinking about the best way of this implementation and as soon as I have something working I shall upload. Keep in mind that we could need the use of a table to store the WiFi SSID and the password (or just keep them in memory untill wpa_supplicant.conf is written).

**Comment by twa127** [comment]
Hi,

Can we have a repository for V2 ?

**Comment by pihome-shc** [comment]
PiHome - Next Generation Smart Thermostat version 2.0 repo is ready 
https://github.com/pihome-shc/PiHomeHVAC

**Comment by twa127** [comment]
Hi,

Some notes on this version

The previous version was centered around the creation of zones, this version is more centered around devices.
The system setup process is

1. Add Nodes (as per the previous version)
2. Create Temperature Measurement devices (with a node and child id)
3. Create Relay devices (with a node and child id)
4. Create Zones using the temperature measurement and relay devices created previously
5. Temperature Measurement devices can be display without the need to be allocated to a 'zombie' zone

The Temperature Measurement devices can be created from the One Touch menu
![deploy1](https://user-images.githubusercontent.com/46624596/97433327-3a6a8880-1915-11eb-93b0-feac36159403.JPG)

New configuration GUIs are available for the Relays and Sensors devices
![deploy2](https://user-images.githubusercontent.com/46624596/97433533-946b4e00-1915-11eb-942b-75e2637affc8.JPG)

Part of the reason for this version is to support HVAC systems and hence the system can be configured to work in either the existing Boiler or HVAC mode. The mode is selected from settings/system configuration menu
![deploy3](https://user-images.githubusercontent.com/46624596/97433816-fe83f300-1915-11eb-9ae8-5b8b38f535ee.JPG)

The home screen will show the mode of operation
![deploy4](https://user-images.githubusercontent.com/46624596/97433953-33904580-1916-11eb-816c-0e33535a6831.JPG)
![deploy5](https://user-images.githubusercontent.com/46624596/97434180-8964ed80-1916-11eb-9ccf-9962900bc8f4.JPG)

The 'MODE' button will cycle through the various modes of operation, for Boiler mode this is OFF, TIMER, CE, HW or BOTH (not yet implemented), for HVAC the modes are OFF, TIMER, AUTO, HEAT, COOL or FAN.

The trusted boiler.php engine has been replaced by controller.php

I have left 2 sample databases in the MySQL_Database directory, eu_27_10_2020.sql and hvac_26_10_2020.sql

Any comments more than welcome

**Comment by sandreialexandru** [comment]
It looks great! Nice work! Looking forward to testing the new version.
Could you please explain in a few words what TIMER, CE and HW(hot water, i believe) are and how are they changed from previous version?

**Comment by twa127** [comment]
Hi 

The idea was a spin off from the HVAC mode, the idea is to provide the same kind of function you get on a standard central heating controller, where you have OFF, Central Heating ON, Hot Water On, Both ON or TIMED. At the moment it does not do anything, I can always drop it, if it is not wanted

**Comment by sandreialexandru** [comment]
How about a Discord server, where we can chat about features and ask questions, leaving Github only for technical discussions?

**Comment by pihome-shc** [comment]
@sandreialexandru i agree, i have receive tons of email where people ask for help, i have to migrate PiHome site from current hosting as it is performing very very slow and i have plane to this on 7th November,

**Comment by sandreialexandru** [comment]
@twa127 I see that you are adding live temp to PiHOME v2 with the mode selector(central heating mode).
I think the modes should be MANUAL(live temp) or TIMED(schedule, boost, override, night climate, holiday) and leave the zone based heating as it was. 
Let's say that Hot Water is the selected mode. What happens with the heating zones? No heating will occur? If Central Heating is selected, then no hot water zones will be managed.
Another idea would be to have a switch under each zone card to toggle between manual and timed, but this would mean that we would have to keep track of the mode for every zone. Doesn't a heating flow based logic make it less complicated? (Boost>Night climate>Override>Schedule>Live temperature)

**Comment by sandreialexandru** [comment]
Searching for ideas of new features got me thinking about OVERRIDE. I see override as boost without a timer cut-off, which as @twa127  said, is a safety feature.  I don't see any use of override as long as it is simple to edit a schedule or to modify target temp with live temperature slider.  Do you agree?

**Comment by sandreialexandru** [comment]
@pihome-shc I have created an open server on Discord, so we cand chat about new features. This can be temporary untill you have time to create one for PiHOME or i can transfer ownership to you. 
Use this invite code:
https://discord.gg/mXKZSsG
